compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -196,6 +197,33 @@ impl ItemLikeVisitor<'tcx> for Collector<'tcx> {
                 .span_label(m.span, "missing `name` argument")
                 .emit();
             }
+
+            if let (NativeLibKind::RawDylib, Some(lib_name)) = (lib.kind, lib.name) {
+                if !self.tcx.sess.target.options.is_like_windows {
+                    sess.span_fatal(
+                        m.span,
+                        "`#[link(...)]` with `kind = \"raw-dylib\"` only supported on Windows",
+                    );
+                }
+                if !self.tcx.sess.target.options.is_like_msvc {
+                    sess.span_warn(
+                        m.span,
+                        "`#[link(...)]` with `kind = \"raw-dylib\"` not supported on windows-gnu",
```

Comment:
In https://github.com/rust-lang/rust/pull/84171#issuecomment-837308837 mati865 says that the current scheme works ok on windows-gnu if lld is used as a linker instead of ld.

Perhaps the warning should say that it's not *always* supported and recommend switching to lld.
You can use something like `!(sess.opts.cg.linker.contains("lld") || sess.opts.cg.linker_flavor.contains("lld"))` as a heuristic to avoid reporting the warning.

Reply:
People usually enable LLD using `-C link-arg=-fuse-ld=lld` so this would still appear. I think it's not worth the trouble.

Reply:
I'm confused about the linker situation on windows-gnu.  In private communication (Zulip, May 3 of this year), @mati865 tells me, "It's not doable to use LLD in Rust CI."  If I understand that correctly, there doesn't seem to be any way to add test coverage of this if we were to declare the feature supported on windows-gnu with the lld linker, and I'm deeply reluctant to add commit this without at least some degree of test coverage.

I'll need to create a follow-on PR with necessary changes to the unstable-book anyway; would it be appropriate to discuss these issues in that book?

Reply:
Ok, let's leave this as is then.

Reply:
Sounds good; thanks.